### PR TITLE
support select for update in plan

### PR DESCRIPTION
**Pull request by YANGGMM**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #9790 

## What this PR does / why we need it:
support select for update in plan

### CODE REVIEW

**Review by aunjgr — APPROVED** [review]

File: pkg/sql/plan/build_test.go [hunk 1]
```diff
@@ -505,6 +506,7 @@ func TestSingleTableSQLBuilder(t *testing.T) {
 
 		"SELECT DISTINCT N_NAME FROM NATION GROUP BY N_REGIONKEY", //test distinct with group by
 		"SELECT DISTINCT N_NAME FROM NATION ORDER BY N_REGIONKEY", //test distinct with order by
+		"select count(n_name) from nation limit 10 for update",
```
Comment (fengttt) [inline pkg/sql/plan/build_test.go:509]: What does this mean?     select update on an aggregate?    We lock everything in nation?

File: pkg/sql/plan/build_test.go [hunk 2]
```diff
@@ -488,6 +488,7 @@ func TestSingleTableSQLBuilder(t *testing.T) {
 
 		"values row(1,1), row(2,2), row(3,3) order by column_0 limit 2",
 		"select * from (values row(1,1), row(2,2), row(3,3)) a (c1, c2)",
+		"select * from nation where n_name like ? or n_nationkey > 10 order by 2 limit '10' for update",
```
Comment (fengttt) [inline pkg/sql/plan/build_test.go:491]: I assume this query we will select primary key out as well and lock it before returning?

File: pkg/sql/plan/build_test.go (hunk 1 above)
Comment (YANGGMM) [reply pkg/sql/plan/build_test.go:509]: no, wo not support select egg now, we will get a err info

File: pkg/sql/plan/build_test.go (hunk 2 above)
Comment (YANGGMM) [reply pkg/sql/plan/build_test.go:491]: yes, but it will have some different when there has a limit

**Pull request merged**